filebeat/harvester/reader/multiline.go

```diff
@@ -113,40 +115,40 @@ func (mlr *Multiline) Next() (Message, error) {
 }
 
 func (mlr *Multiline) readFirst() (Message, error) {
-	for {
-		message, err := mlr.reader.Next()
-		if err != nil {
-			// no lines buffered -> ignore timeout
-			if err == sigMultilineTimeout {
-				continue
-			}
+	//for {
+	message, err := mlr.reader.Next()
+	if err != nil {
+		// no lines buffered -> ignore timeout
+		//if err == sigMultilineTimeout {
+		//	continue
+		//}
 
-			// pass error to caller (next layer) for handling
-			return message, err
-		}
+		// pass error to caller (next layer) for handling
+		return message, err
+	}
 
-		if message.Bytes == 0 {
-			continue
-		}
+	//if message.Bytes == 0 {
+	//	continue
+	//}
 
-		// Start new multiline event
-		mlr.clearBuffer()
-		mlr.resetBuffer(message)
-		mlr.setState((*Multiline).readNext)
-		return mlr.readNext()
-	}
+	// Start new multiline event
+	mlr.clearBuffer()
+	mlr.resetBuffer(message)
+	mlr.setState((*Multiline).readNext)
+	return mlr.readNext()
+	//}
 }
 
 func (mlr *Multiline) readNext() (Message, error) {
 	for {
 		message, err := mlr.reader.Next()
 		if err != nil {
 			// handle multiline timeout signal
-			if err == sigMultilineTimeout {
+			if err == sigMultilineTimeout || err == io.EOF {
```

Comment:
About io.EOF: As suggested in the PR description, I suggest timeout to be 0 by default which has kind of the same affect. It is based on the assumption that one multiline event is flushed at once to disk. 

You are right that the log_file reader does not return the io.EOF by default. It is returned if the reader is not Continutable or CloseEOF is set. 

The problem about setting timeout to 0 is that it will directly return the signal after the first line it read. Instead it should only send a signal if there is no more data.